spec/std/db/dummy_driver.cr

```diff
@@ -0,0 +1,181 @@
+require "spec"
+
+class DummyDriver < DB::Driver
+  def build_connection(db : DB::Database) : DB::Connection
+    DummyConnection.new(db)
+  end
+
+  class DummyConnection < DB::Connection
+    def initialize(db)
+      super(db)
+      @@connections ||= [] of DummyConnection
```

Comment:
We still can remove `.not_nil!`, if we will define `self.connections` as:

```crystal
def self.connections
  @@connections ||= [] of DummyConnection
end
```

and use everywhere else `self.connection` instead of `@@connections`.